x-pack/plugins/apm/public/components/alerting/rule_types/register_apm_rule_types.ts

```diff
@@ -148,11 +149,29 @@ export function registerApmRuleTypes(
       return `${docLinks.links.alerting.apmRules}`;
     },
     ruleParamsExpression: lazy(() => import('./anomaly_rule_type')),
-    validate: () => ({
-      errors: [],
-    }),
+    validate: validateAnomalyRule,
     requiresAppContext: false,
     defaultActionMessage: anomalyMessage,
     priority: 90,
   });
 }
+
+function validateAnomalyRule(ruleParams: AlertParams) {
+  const validationResult = { errors: {} };
+  const errors: {
+    anomalyDetectorTypes?: string;
+  } = {};
+  validationResult.errors = errors;
```

Comment:
Why is this written like this?

Reply:
I follow how validation was done in infra rules